java/navigate-route/README.md

```diff
@@ -2,33 +2,32 @@
+Tap 'Navigate' to simulate traveling and to receive directions from a preset starting point to a preset destination. Tap 'Reset' to start the simulation from the beginning.
 
 ## How it works
 
 1. Create a `RouteTask` using a URL to an online route service.
-2. Generate default `RouteParameters`.
-4. Add `Stop`s to the parameters.
-5. Solve the route to get a `RouteResult`.
+2. Generate default `RouteParameters` using `routeTask.createDefaultParametersAsync()`.
+3. Set `returnStops`, `returnDirections`, and `returnRoutes` on the parameters to true.
+4. Add `Stop`s to the parameters `stops` collection for each destination.
+5. Solve the route using `routeTask.solveAsync(routeParameters)` to get a `RouteResult`.
 6. Create a `RouteTracker` using the route result, and the index of the desired route to take.
-7. Use `trackLocationAsync(LocationDataSource.Location)` to track the location of the device and update the route tracking status.
-8. Add a listener to capture `TrackingStatusChangedEvent`s, then get the `TrackingStatus` and use it to display updated route information. Tracking status includes a variety of information on the route progress, such as the remaining distance, remaining geometry or traversed geometry (represented by a `Polyline`), or the remaining time in minutes (`Double`), amongst others.
-9. Add a `NewVoiceGuidanceListener` to get `VoiceGuidance` whenever new instructions are available. From the voice guidance, get the `String` representing the directions and use a text-to-speech engine to output the maneuver directions.
-10. You can also query the tracking status for the current `DirectionManeuver` index, retrieve that maneuver from the `Route` and get it's direction text to display in the GUI.
-11. To establish whether the destination has been reached, get the `DestinationStatus` from the tracking status. If the destination status is `REACHED`, and the `remainingDestinationCount` is 1, we have arrived at the destination and can stop routing. If there are several destinations in your route, and the remaining destination count is greater than 1, switch the route tracker to the next destination.
+7. Use `routeTracker.trackLocationAsync(LocationDataSource.Location)` to track the location of the device and update the route tracking status.
+8. Add a listener to capture `LocationChangedEvent`s from `MapView.LocationDisplay`, and then get the `TrackingStatus` and use it to display updated route information. Tracking status includes a variety of information on the route progress, such as the remaining distance, remaining geometry or traversed geometry (represented by a `Polyline`), or the remaining time (`Double`), amongst others.
+9. Add a `NewVoiceGuidanceListener` to get the `VoiceGuidance` whenever new instructions are available. From the voice guidance, get the `String` representing the directions and use a text-to-speech engine to output the maneuver directions.
```

Comment:
these should be referred to as methods rather than as classes e.g. addNewVoiceGuidanceListener and getVoiceGuidance

Reply:
Prior to this suggestion I rewrote step 8 in a manner similar to step 9. So I tried to rewrite both. I found this kind of challenging. If you have suggestions, I'd certainly use them.